### PR TITLE
Introduce UUID for Beat to uniquely identify each beat

### PR DESCRIPTION
**Pull request by ruflin**
This is a required for centralised configuration management. See #464
For the UUID id the goal is follow rfc4122 standrard: http://www.ietf.org/rfc/rfc4122.txt

### CODE REVIEW

**Comment by ruflin** [comment]
@urso @andrewkroh I added the glide.lock file to versioning. Please let me know in case this was left out on purpose.

**Comment by ruflin** [comment]
@urso @andrewkroh After not managing to get it build on travis with a new glide dependency I decided to implement the UUID part myself. I would still like to understand what went wrong with my glide changes. It worked locally but on Travis I got these errors: https://travis-ci.org/ruflin/beats/jobs/99585703#L226

**Comment by tsg** [comment]
@ruflin maybe you forgot to delete the .git folder before adding the new dep sources? Glide is different from godep in the sense that it leaves the .git folder there, and then git thinks it is a submodule.

**Comment by ruflin** [comment]
@tsg I think that was actually the issue. I assumed when I use `glide get ...` it would do such magic for me.

**Comment by urso** [comment]
@ruflin @tsg that's cause glide is kinda modelled after bundler and kinda discourages you to have your vendor directory checked in, but rely on the lock file.

File: libbeat/common/uuid.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package common
+
+import (
+	"crypto/rand"
+	"fmt"
+)
+
+// GenerateUUID is used to generate a random UUID
+// This was copied from https://github.com/hashicorp/uuid/blob/master/uuid.go
```
Comment (urso) [inline libbeat/common/uuid.go]: copied code license is Mozilla License.

Please let's not copy others code and either.

We want RFC compatible UUIDs? If so, which uuid version is this? Looks like 4 (all random), but missing version number.

Comment (ruflin) [inline libbeat/common/uuid.go]: Yes, the intention is to use UUID version 4.

Comment (ruflin) [inline libbeat/common/uuid.go]: @urso Should we go with an external lib with a better license or our own implementation? As a first step I would suggest to pick an other external implementation and bring it in house if needed later. Agree?

Comment (tsg) [inline libbeat/common/uuid.go]: If we can find a library that can make use of host and/or time information to guarantee uniqueness, I'm +1 on using that.

Comment (ruflin) [inline libbeat/common/uuid.go]: @tsg I think that is more a discussion about which version of UUID we want to use. Host and time sounds like version 1 UUID. I prefer version 4 as it is fully random and does not have any dependencies on time or host. I would not use the UUID to try to figure out host and time. If this information is needed, I would send it separately.

Comment (ruflin) [inline libbeat/common/uuid.go]: @tsg I will suggest a library that supports all UUID versions so it is also easy to switch.

**Comment by ruflin** [comment]
@urso @tsg I added now the satori/go.uuid lib which supports UUID v1-v5. If this looks good, I would rebase again.

File: libbeat/common/uuid.go (hunk 1 above)
Comment (tsg) [inline libbeat/common/uuid.go]: +1. I meant host and time as name spacing to reduce or eliminate the chances of conflicts.

Comment (urso) [inline libbeat/common/uuid.go]: when using UUID-4 you'd want a really good seed with enough entropy + pseudo random number generate to not increase chance of collisions. If you can not fully guarantee this, UUID 5 (basically a hash value) is a viable option.

**Comment by tsg** [comment]
LGTM.

File: libbeat/beat/beat_test.go [hunk 2]
```diff
@@ -13,6 +14,19 @@ func Test_NewBeat(t *testing.T) {
 
 	assert.Equal(t, "testbeat", b.Name)
 	assert.Equal(t, "0.9", b.Version)
+
+	// UUID4 should be 36 chars long
+	assert.Equal(t, 16, len(b.UUID))
+	assert.Equal(t, 36, len(b.UUID.String()))
+}
+
+func Test_NewBeat_UUI(t *testing.T) {
```
Comment (mrkschan) [inline libbeat/beat/beat_test.go]: Is this a typo?

Comment (ruflin) [inline libbeat/beat/beat_test.go]: Yes. Thanks for spotting.

Comment (ruflin) [inline libbeat/beat/beat_test.go]: Fixed here: https://github.com/ruflin/beats/commit/2bbd9e842199a78b4eb18485beb5a4394e5c88e3

**Pull request merged**